hudi-common/src/main/java/org/apache/hudi/common/engine/RecordContext.java

```diff
@@ -70,6 +70,15 @@ protected RecordContext(HoodieTableConfig tableConfig) {
         .orElseThrow(() -> new IllegalArgumentException("No record keys specified and meta fields are not populated")));
   }
 
+  /**
+   * Constructs an instance of {@link RecordContext} that provides field accessor methods but cannot compute the record key for records.
+   */
+  protected RecordContext() {
```

Comment:
Can we update the constructor with `TypeConverter` as parameter and initialize `typeConverter` to avoid potential NPE?

Reply:
Yes, updating now